sympy/parsing/c/c_parser.py

```diff
@@ -1033,6 +1033,30 @@ def get_expr_for_operand(self, combined_variable):
             if combined_variable[1] == 'boolean':
                     return true if combined_variable[0] == 'true' else false
 
+        def transform_while_stmt(self, node):
+            """Transformation function for handling while statement
+
+            Returns
+            =======
+
+            statement : Codegen AST Node
+                contains the while statement node
+
+            """
+            children = node.get_children()
+
+            condition = self.transform(next(children))
+            statements = self.transform(next(children))
```

Comment:
what happens if the while statement is empty `while(i>0);` and if there is no condition `while() ...`.

Can you add some tests for these and some failsafe in case these don't return anything or throw an error.

Reply:
If there is no condition, clang will not parse even while statement. So, there is no need to worry about that.

But yeah, if the statement block is empty, null_stmt will be returned, which is not yet supported by our code (will get ignored). So, None will be passed inside the CodeBlock part, which will result in the correct parsing! But, we can define a function to support null_stmt explicitly return None, which will be better.

Reply:
Currently, on this branch:
#### Input:
```
void fn() {
    int i=0;
    while(1);
}
```
#### Output:
![while](https://user-images.githubusercontent.com/38134283/95014750-8bfd5b80-0666-11eb-8989-50265d4e4703.png)

This looks good, right! (Though it is much better to explicitly pass None in Codeblock)

Reply:
Looks alright. But, as you said, Explicit is better than implicit(PEP 20). 

Also, can you add a test for this?